src/overlays/actors/ovl_Bg_Haka_Trap/z_bg_haka_trap.c

```diff
@@ -40,36 +41,490 @@ const ActorInit Bg_Haka_Trap_InitVars = {
+    cosine = Math_Coss(this->dyna.actor.shape.rot.y);
+    if (this->dyna.actor.params == TRAP_GUILLOTINE_SLOW) {
+        sp28.x = CLAMP(sp28.x, -50.0f, 50.0f);
+        zNonNegative = (sp28.z >= 0.0f) ? 1.0f : -1.0f;
+        sp28.z = zNonNegative * -15.0f;
+    } else {
+        sp28.x = -CLAMP(sp28.x, -162.0f, 162.0f);
+        zNonNegative = (sp28.z >= 0.0f) ? 1.0f : -1.0f;
+        sp28.z = zNonNegative * 15.0f;
+    }
+
+    this->colliderCylinder.dim.pos.x = (s16)((this->dyna.actor.posRot.pos.x + (sp28.x * cosine)) + (sp28.z * sine));
+    this->colliderCylinder.dim.pos.z = (s16)((this->dyna.actor.posRot.pos.z + (sp28.x * sine)) + (sp28.z * cosine));
+}
+
+void func_808801B8(BgHakaTrap* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s32 temp = 4;
+
+    if ((D_80880F30 == 0) && (func_8008E988(globalCtx) == 0)) {
+        if (!Math_ApproxF(&this->dyna.actor.posRot.pos.x, this->dyna.actor.initPosRot.pos.x, 0.5f)) {
+            func_8002F974(&this->dyna.actor, 0x2058);
+        } else if (this->dyna.actor.params == TRAP_SPIKED_WALL) {
+            D_80881018 |= 1;
+        } else if (this->dyna.actor.params == TRAP_SPIKED_WALL_2) {
+            D_80881018 |= 2;
+        }
+    }
+
+    func_8087FFC0(this, globalCtx);
+
+    if ((this->colliderSpikes.base.acFlags & 2) != 0) {
+        this->actionTimer = 20;
+        D_80880F30 = 1;
+        this->actionFunc = func_808802D8;
+    } else if (D_80881018 == 3) {
+        D_80881018 = temp;
+        player->actor.bgCheckFlags |= 0x100;
+    }
+}
+
+void func_808802D8(BgHakaTrap* this, GlobalContext* globalCtx) {
+    Vec3f vector;
+    f32 temp;
+    s32 i;
+
+    if (this->actionTimer != 0) {
+        this->actionTimer--;
+    }
+
+    func_8002F974(&this->dyna.actor, 0x205B);
+
+    for (i = 0; i < 2; i++) {
+        f32 rand = Math_Rand_ZeroOne();
+
+        temp = (this->dyna.actor.params == TRAP_SPIKED_WALL) ? -30.0f : 30.0f;
+
+        vector.x = temp * rand + this->dyna.actor.posRot.pos.x;
+        vector.y = Math_Rand_ZeroOne() * 10.0f + this->dyna.actor.posRot.pos.y + 30.0f;
+        vector.z = Math_Rand_CenteredFloat(320.0f) + this->dyna.actor.posRot.pos.z;
+
+        func_8002A6B8(globalCtx, &vector, &D_8088101C, &D_8088101C, 0x82, 0x14, 0xFF, 0xFF, 0x96, 0xAA, 0xFF, 0, 0, 1,
+                      9, 0);
+    }
+
+    if (this->actionTimer == 0) {
+        D_80880F30 = 0;
+        Actor_Kill(&this->dyna.actor);
+    }
+}
+
+void func_80880484(BgHakaTrap* this, GlobalContext* globalCtx) {
+    s32 sp24;
+    s32 temp;
+
+    if (this->unk_16A) {
+        this->dyna.actor.velocity.y *= 3.0f;
+    } else {
+        this->dyna.actor.velocity.y *= 2.0f;
+    }
+
+    if (this->actionTimer != 0) {
+        this->actionTimer -= 1;
+    }
+
+    sp24 = Math_ApproxF(&this->dyna.actor.posRot.pos.y, this->dyna.actor.initPosRot.pos.y - 185.0f,
+                        this->dyna.actor.velocity.y);
+    temp = this->actionTimer;
```

Comment:
unnamed variables are fine, but in this case when its very obvious what it is i think it would be good to name this "timer"

Reply:
fixed